Source/DafnyCore/Verifier/ProofObligationDescription.cs

```diff
@@ -1109,15 +1109,20 @@ public class AssignmentShrinks : ProofObligationDescriptionWithNoExpr {
 
 public class BoilerplateTriple : ProofObligationDescriptionWithNoExpr {
   public override string SuccessDescription =>
-    $"error is impossible: {msg}";
+    comment + (successMessage is not null ? $" ({successMessage})" : "");
 
-  public override string FailureDescription => msg;
+  public override string FailureDescription =>
+    comment + (errorMessage is not null ? $" ({errorMessage})" : "");
 
   public override string ShortDescription => "boilerplate triple";
 
-  private readonly string msg;
+  private readonly string errorMessage;
+  private readonly string successMessage;
+  private readonly string comment;
```

Comment:
There is a class named "ProofObligationDescriptionCustomMessages" that you can use to implement custom success and failure messages, so that we have a consistent interface. Could you make use it? I see you have a comment as well there, feel free to override and use comment + base.SuccessMessage

Reply:
Ah, thanks for pointing that out! I hadn't noticed that you'd added that. :)